src/geometries/point.ts

```diff
@@ -22,53 +23,22 @@ export class Point extends BaseGeometry<GeoJSONPoint> {
    * @param metatile size of a metatile
    * @returns tile within the tile matrix
    */
-  public toTile<T extends TileMatrixSet>(tileMatrixSet: T, tileMatrixId: TileMatrixId<T>, reverseIntersectionPolicy: boolean, metatile = 1): Tile<T> {
+  public toTile<T extends TileMatrixSet>(
+    tileMatrixSet: T,
+    tileMatrixId: TileMatrixId<T>,
+    reverseIntersectionPolicy: ReverseIntersectionPolicy,
+    metatile = 1
+  ): Tile<T> {
     validateMetatile(metatile);
     // validateTileMatrixSet(tileMatrixSet); // TODO: missing implementation
     validateCRSByOtherCRS(this.coordRefSys, tileMatrixSet.crs);
     validateTileMatrixIdByTileMatrixSet(tileMatrixId, tileMatrixSet);
 
-    const tileMatrix = tileMatrixSet.getTileMatrix(tileMatrixId);
-    if (!tileMatrix) {
-      throw new Error('tile matrix id is not part of the given tile matrix set');
-    }
-
-    validatePointByTileMatrix(this, tileMatrix);
-
-    const [east, north] = this.coordinates;
-
-    const width = tileEffectiveWidth(tileMatrix) * metatile;
-    const height = tileEffectiveHeight(tileMatrix) * metatile;
-
-    const [tileMatrixBoundingBoxMinEast, tileMatrixBoundingBoxMinNorth, tileMatrixBoundingBoxMaxEast, tileMatrixBoundingBoxMaxNorth] =
-      tileMatrixToBBox(tileMatrix);
-    const { cornerOfOrigin = 'topLeft' } = tileMatrix;
-
-    const tempTileCol = (east - tileMatrixBoundingBoxMinEast) / width;
-    const tempTileRow = (cornerOfOrigin === 'topLeft' ? tileMatrixBoundingBoxMaxNorth - north : north - tileMatrixBoundingBoxMinNorth) / height;
-
-    // when explicitly asked to reverse the intersection policy (location on the edge of the tile)
-    if (reverseIntersectionPolicy) {
-      const onEdgeEastTranslation = east === tileMatrixBoundingBoxMinEast ? 1 : 0;
-      const onEdgeNorthTranslation = north === (cornerOfOrigin === 'topLeft' ? tileMatrixBoundingBoxMaxNorth : tileMatrixBoundingBoxMinNorth) ? 1 : 0;
-
-      const tileCol = Math.ceil(tempTileCol) - 1 + onEdgeEastTranslation;
-      const tileRow = Math.ceil(tempTileRow) - 1 + onEdgeNorthTranslation;
-
-      return new Tile({ col: tileCol, row: tileRow, tileMatrixId }, tileMatrixSet, metatile);
-    }
-
-    // when east/north is on the maximum edge of the tile matrix (e.g. lon = 180 lat = 90 in wgs84)
-    const onEdgeEastTranslation = east === tileMatrixBoundingBoxMaxEast ? 1 : 0;
-    const onEdgeNorthTranslation = north === (cornerOfOrigin === 'topLeft' ? tileMatrixBoundingBoxMinNorth : tileMatrixBoundingBoxMaxNorth) ? 1 : 0;
-
-    const tileCol = Math.floor(tempTileCol) - onEdgeEastTranslation;
-    const tileRow = Math.floor(tempTileRow) - onEdgeNorthTranslation;
-
-    return new Tile({ col: tileCol, row: tileRow, tileMatrixId }, tileMatrixSet, metatile);
+    const { col, row } = positionToTileIndex(this.coordinates, tileMatrixSet, tileMatrixId, reverseIntersectionPolicy, metatile);
```

Comment:
You are sending `coordinates` but expecting a `position` with `east` and `north` points, does this match up? Is it only because it's a `point`?

Reply:
the types match for a good reason
the class uses a geojson point generic type